src/stores/ForeignStore.js

```diff
@@ -323,6 +335,43 @@ class ForeignStore {
       console.error(e)
     }
   }
+
+  async getFeeEvents() {
+    try {
+      const { FOREIGN_ABI } = getBridgeABIs(this.rootStore.bridgeMode)
+      const abi = [...FOREIGN_ABI, ...BaseFeeManager]
+      const contract = new this.foreignWeb3.eth.Contract(abi, this.FOREIGN_BRIDGE_ADDRESS);
+      const deployedAtBlock = await getDeployedAtBlock(this.foreignBridge);
+      const events = await getPastEvents(contract, deployedAtBlock, 'latest')
+
+      processLargeArrayAsync(
+        events,
+        this.processEvent,
+        () => {
+          this.feeEventsFinished = true
+        })
+    } catch(e){
+      console.error(e)
+      this.getFeeEvents()
+    }
+  }
+
+  processEvent = (event) => {
+    if (event.event === "FeeDistributedFromSignatures") {
```

Comment:
I think we need to cover a case to total fees from `FeeDistributedFromAffirmation` on the Foreign side as well. I could image fee managers for `native-to-erc` or `erc-to-erc` (where both tokens are erc677) bridge modes when the fees could be distributed on the foreign side before validators actually relay the request to another side.

Reply:
Thanks for suggestion. Added here https://github.com/poanetwork/bridge-ui/pull/211/commits/a9df93f6c27adf405f7046562f86691483cd02b3